mico-core/src/main/java/io/github/ust/mico/core/ServiceRepository.java

```diff
@@ -13,4 +13,6 @@
     Service findByName(String name, @Depth int depth);
 
     Service findByShortName(String shortName, @Depth int depth);
+
+    //TODO: If multiple versions exist for one Service, the methods return "Incorrect result size: expected at most 1"
```

Comment:
We need to think, how we want to fix this.
```suggestion
package io.github.ust.mico.core;

import org.springframework.data.neo4j.annotation.Depth;
import org.springframework.data.neo4j.repository.Neo4jRepository;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface ServiceRepository extends Neo4jRepository<Service, Long> {

    List<Service> findByName(@Param("name") String name);

    List<Service> findByShortName(@Param("shortName") String shortName);

    List<Service> findByName(@Param("name") String name, @Depth int depth);

    List<Service> findByShortName(@Param("shortName") String shortName, @Depth int depth);

    Service findByNameAndVersion(@Param("name") String name, @Param("version") String version);

    Service findByShortNameAndVersion(@Param("shortName") String shortName, @Param("version") String version);

    Service findByNameAndVersion(@Param("name") String name, @Param("version") String version, @Depth int depth);

    Service findByShortNameAndVersion(@Param("shortName") String shortName, @Param("version") String version, @Depth int depth);
}
```